cmd/harmony/main.go

```diff
@@ -699,6 +708,37 @@ func setupPrometheusService(node *node.Node, hc harmonyConfig, sid uint32) {
 	node.RegisterService(service.Prometheus, p)
 }
 
+func setupSyncService(node *node.Node, host p2p.Host, hc harmonyConfig) {
+	blockchains := []*core.BlockChain{
```

Comment:
what if it's the beacon chain, then there is only one chain that needs sync

Reply:
Blockchains will be de-duplicated in `NewDownloaders` (hmy/downloader/downloaders.go:14), so finally there will only be one downloader running. 

But it seems this piece of code is causing confusion, let me refactor it a little bit.